rent_golem/backend/rent_golem/clusters/tests.py

```diff
@@ -1,3 +1,225 @@
+            "owner": self.owner.id
+        }
+
+        response = self.client.post(reverse('cluster-list'), data=cluster_data, format="json")
+
+        self.assertEquals(response.status_code, status.HTTP_401_UNAUTHORIZED)
+        self.assertFalse(Cluster.objects.filter(uuid=cluster_data["uuid"]).exists())
+
+    def test_read_one_object_from_db_as_registered_user(self):
+        cluster = self.create_cluster_object(owner=self.owner)
+
+        self.client.force_authenticate(user=self.owner)
+
+        get_response = self.client.get(reverse('cluster-detail', args=(cluster.uuid,)))
+
+        self.assertEquals(get_response.status_code, status.HTTP_200_OK)
+        self.assertEquals(get_response.data["package_type"], cluster.package_type)
+        self.assertEquals(get_response.data["status"], cluster.status)
+        self.assertEquals(get_response.data["additional_params"], cluster.additional_params)
+        self.assertEquals(get_response.data["size"], cluster.size)
+        self.assertEquals(get_response.data["owner"], self.owner.id)
+
+    def test_read_list_of_3_db_objects_as_registered_user(self):
+        uuids = {
+            "acec5482-59ef-4c0e-9ea4-4e5451c3cbda",
+            "bcec5482-59ef-4c0e-9ea4-4e5451c3cbda",
+            "ccec5482-59ef-4c0e-9ea4-4e5451c3cbda"
+        }
+        for uuid in uuids:
+            self.create_cluster_object(uuid=uuid, owner=self.owner)
+
+        jwt = self.client.post(
+            reverse("token_obtain_pair"), data={"email": self.owner.email, "password": "test1234"}
+        ).data["access"]
+
+        list_response = self.client.get(reverse('cluster-list'), headers={"Authorization": f"Bearer {jwt}"})
+
+        self.assertEquals(list_response.status_code, status.HTTP_200_OK)
+        response_uuids = {list_element["uuid"] for list_element in list_response.data}
+        self.assertSetEqual(response_uuids, uuids)
+
+    def test_read_list_of_3_db_objects_as_anonymous_user(self):
+        uuids = {
+            "acec5482-59ef-4c0e-9ea4-4e5451c3cbda",
+            "bcec5482-59ef-4c0e-9ea4-4e5451c3cbda",
+            "ccec5482-59ef-4c0e-9ea4-4e5451c3cbda"
+        }
+        for uuid in uuids:
+            self.create_cluster_object(uuid=uuid, owner=self.owner)
+
+        list_response = self.client.get(reverse('cluster-list'))
+
+        self.assertEquals(list_response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    # Update
+    def test_update_by_owner(self):
+        cluster = self.create_cluster_object(owner=self.owner)
+
+        jwt = self.client.post(
+            reverse("token_obtain_pair"), data={"email": self.owner.email, "password": "test1234"}
+        ).data["access"]
+
+        update_response = self.client.patch(
+            reverse('cluster-detail', args=(cluster.uuid,)),
+            data={'size': cluster.size + 1},
+            headers={"Authorization": f"Bearer {jwt}"}
+        )
+
+        self.assertEquals(update_response.status_code, status.HTTP_202_ACCEPTED)
```

Comment:
If successful PATCH returns data then let's return HTTP_200_OK, if nothing is returned let's return HTTP_204_NO_CONTENT